### PR TITLE
Make typesToList read metaTypesNotToList from new p.a.registry settings.

### PR DESCRIPTION
**Pull request by tisto**

### CODE REVIEW

File: Products/CMFPlone/utils.py [hunk 1]
```diff
@@ -234,15 +234,15 @@ def getEmptyTitle(context, translated=True):
...
-    bl = ntp.getProperty('metaTypesNotToList', ())
-    bl_dict = {}
-    for t in bl:
-        bl_dict[t] = 1
-    all_types = ttool.listContentTypes()
-    wl = [t for t in all_types if not t in bl_dict]
-    return wl
+    from zope.component import getUtility
+    from plone.registry.interfaces import IRegistry
+    from Products.CMFPlone.interfaces import INavigationSchema
```
Comment (jensens) [inline Products/CMFPlone/utils.py]: I think its fine to import these in the files top

Comment (tisto) [inline Products/CMFPlone/utils.py]: This is work in progress. Tests failed after moving imports. I need to investigate...

**Comment by tisto** [comment]
http://jenkins.plone.org/job/pull-request/109/console passes

File: Products/CMFPlone/utils.py [hunk 2]
```diff
@@ -234,15 +237,13 @@ def getEmptyTitle(context, translated=True):
...
-    ttool = getToolByName(context, 'portal_types')
-    bl = ntp.getProperty('metaTypesNotToList', ())
-    bl_dict = {}
-    for t in bl:
-        bl_dict[t] = 1
-    all_types = ttool.listContentTypes()
-    wl = [t for t in all_types if not t in bl_dict]
-    return wl
+    registry = getUtility(IRegistry)
+    from Products.CMFPlone.interfaces import INavigationSchema
```
Comment (jensens) [inline Products/CMFPlone/utils.py]: so this creates a circular import problem in the header.
I looked up why:
in https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/interfaces/controlpanel.py#L4 we import from utils a json validator. this validator is used *only* in ``interfaces/controlpanel``. I propose to either move the validator out of utils into ``plone.app.z3cform`` - we have already validations defined there and json validation is very generic.

**Comment by tisto** [comment]
Jenkins is green: http://jenkins.plone.org/job/pull-request/110/

**Comment by tisto** [comment]
plone.app.contentlisting fix: https://github.com/plone/plone.app.contentlisting/pull/7 (though, I'm not sure where this is actually used).

File: Products/CMFPlone/utils.py (hunk 2 above)
Comment (tisto) [inline Products/CMFPlone/utils.py]: We are about to prepare the beta2 release. I will merge this pr now to get the fix in. We can work on that later. Can you point me to the plone.app.z3cform validators?

**Pull request merged**